src/coreclr/src/pal/src/misc/jitsupport.cpp

```diff
@@ -22,7 +22,15 @@ PAL_GetJitCpuCapabilityFlags(CORJIT_FLAGS *flags)
     _ASSERTE(flags);
 
     CORJIT_FLAGS &CPUCompileFlags = *flags;
+
+
 #if defined(HOST_ARM64)
+
+    // Enable ARM64 based flags by default so we always crossgen
+    // ARM64 intrinsics for Linux
+    CPUCompileFlags.Set(InstructionSet_ArmBase);
+    CPUCompileFlags.Set(InstructionSet_AdvSimd);
+
```

Comment:
The check further down, https://github.com/dotnet/runtime/pull/38060/files#diff-f808f893e0b21b8950f8284dd17cdc08L97-L98, should be updated to failfast if `AdvSimd` isn't actually supported, correct?

Reply:
What is the best way to add fail fast inside pal? Looking around I see that `abort` is being used in many files. I could just do `printf/abort`? @davidwrighton - can you please confirm?

```cpp
#ifdef HWCAP_ASIMD
    if ((hwCap & HWCAP_ASIMD) == 0)
    {
        printf("AdvSimd is not supported on the processor.\n");
        abort();
    }
#endif
```

Reply:
Also, I had to move setting `ArmBase` and `AdvSimd` inside `#ifdef (HOST_ARM64)` because I was getting errors:

```
 /__w/1/s/src/coreclr/src/pal/src/misc/jitsupport.cpp:28:25: error: use of undeclared identifier 'InstructionSet_ArmBase'; did you mean 'InstructionSet_X86Base'?
      CPUCompileFlags.Set(InstructionSet_ArmBase);
                          ^~~~~~~~~~~~~~~~~~~~~~
                          InstructionSet_X86Base
```

Reply:
Those should be placed within a #ifdef TARGET_ARM64  *NOT* HOST_ARM64.

Also, with the failfast, I believe abort() is fine. But please print to stderr using fprintf. @janvorli Please verify that for me.

Reply:
@davidwrighton, @kunalspathak it seems a bit strange that a method to query CPU capabilities would assert if it doesn't find the required caps. From the code concern point of view I would rather expect the `EEJitManager::SetCpuInfo` to do the check after the call to the `PAL_GetJitCPUCapabilityFlags` and fail fast using `EEPOLICY_HANDLE_FATAL_ERROR` if the capability is missing. 

Related to this PR, I have noticed there is a (pre-existing) duplication of calls to 
```
    CPUCompileFlags.Set64BitInstructionSetVariants();
    CPUCompileFlags.EnsureValidInstructionSetSupport();
```
which happens both in the PAL helper and in the runtime caller. We can remove the one in PAL, so it would happen in `EEJitManager::SetCpuInfo` for all OSes / archs at single place.

Reply:
Thanks @janvorli for the feedback. 

> expect the EEJitManager::SetCpuInfo to do the check after the call to the PAL_GetJitCPUCapabilityFlags and fail fast using EEPOLICY_HANDLE_FATAL_ERROR if the capability is missing.

So are you suggesting to retrieve caps again using `getauxval()` (for Unix) inside `EEJitManager::SetCpuInfo` and move that check from `PAL_GetJitCPUCapabilityFlags()` to `EEJitManager::SetCpuInfo`? 

```cpp
#if defined(TARGET_UNIX)
    PAL_GetJitCpuCapabilityFlags(&CPUCompileFlags);

#if defined(HOST_ARM64)
#if HAVE_AUXV_HWCAP_H
#ifdef HWCAP_ASIMD
   unsigned long hwCap = getauxval(AT_HWCAP);
   if ((hwCap & HWCAP_ASIMD) == 0)
    {
       EEPOLICY_HANDLE_FATAL_ERROR(COR_E_EXECUTIONENGINE);
    }
#endif // HWCAP_ASIMD
#endif // HAVE_AUXV_HWCAP_H
#endif // HOST_ARM64
#elif defined(HOST_64BIT)
    // FP and SIMD support are enabled by default
...
```
I think there are lot of `#ifdef` to introduce making code less readable just to have a single check, so perhaps keeping it inside `PAL_GetJitCpuCapabilityFlags()` would be better. Let me know if I didn't understand your suggestion or if you disagree.

> We can remove the one in PAL, 

I will send a separate PR to remove redundant pair of `Set64BitInstructionSetVariants(), EnsureValidInstructionSetSupport()` calls from PAL.

Reply:
No, that was not what I meant. My idea was to do this in the codeman.cpp:
```c++
#if defined(TARGET_UNIX)
    PAL_GetJitCpuCapabilityFlags(&CPUCompileFlags);
    if (!CPUCompileFlags.IsSet(InstructionSet_AdvSimd))
    {
        EEPOLICY_HANDLE_FATAL_ERROR_WITH_MESSAGE(COR_E_EXECUTIONENGINE, W("AdvSimd is not supported on the processor."));
    }
#elif defined(HOST_64BIT)
```

and then modify the `PAL_GetJitCpuCapabilityFlags` to not to set `InstructionSet_AdvSimd` unconditionally, but only if (hwCap & HWCAP_ASIMD).